### ISSUE TITLE
if sso_login_url contains ``?``,  login request url can not be built

### ISSUE THREAD

**Issue by gitawego** [Clever/saml2#46]
with SAML of google:
```
https://accounts.google.com/o/saml2/idp?idpid=[your_id]
```
the method ``url.format(uri)`` just ignore completely ``query`` object, if ``search`` exists in object ``uri``, and returns the same url as ``sso_login_url``

I've to parse the ``search`` field, and append to ``query`` object, then remove ``search`` in ``uri``

the url.format works as intended in nodejs as cited in [doc](https://nodejs.org/api/url.html)

     search will be used in place of query.

this special case should be handled in saml2's code.

**Comment by gitawego** [comment]
the temp solution for me is adding this code at line 405 of saml2.coffee:

```
if uri.search
  search = uri.search.substr(1)
  search.split('&').forEach (item) ->
    'use strict'
    m = item.split('=')
    uri.query[m[0]] = m[1]
    return
  delete uri.search
```

**Comment by chad3814** [comment]
this should've been fixed with #56 @jefff release v1.4.4 with it. without #58 though, you won't be able to verify the signature (and if you don't verify the signature anyone can post to your assert url and it'll look like google authorized them)

**Comment by cozmo** [comment]
Fixed in #79 - Published on NPM as `1.8.1`.
